### ISSUE TITLE
`kubectl create deployment --help` needs to be improved 

### ISSUE THREAD

**Issue by Ritikaa96** [kubernetes/kubectl#1457]
**What would you like to be added**:
`kubectl create deployment --help` need improvements

**Why is this needed**:
output of command is 
```
Create a deployment with the specified name.

Aliases:
deployment, deploy

Examples:
  # Create a deployment named my-dep that runs the busybox image
  kubectl create deployment my-dep --image=busybox
  
  # Create a deployment with a command
  kubectl create deployment my-dep --image=busybox -- date
  
  # Create a deployment named my-dep that runs the nginx image with 3 replicas
  kubectl create deployment my-dep --image=nginx --replicas=3
  
  # Create a deployment named my-dep that runs the busybox image and expose port 5701
  kubectl create deployment my-dep --image=busybox --port=5701

Options:
    --allow-missing-template-keys=true:
	If true, ignore any errors in templates when a field or map key is missing in the template. Only applies to
	golang and jsonpath output formats.

    --dry-run='none':
	Must be "none", "server", or "client". If client strategy, only print the object that would be sent, without
	sending it. If server strategy, submit server-side request without persisting the resource.

    --field-manager='kubectl-create':
	Name of the manager used to track field ownership.

    --image=[]:
	Image names to run.

    -o, --output='':
	Output format. One of: (json, yaml, name, go-template, go-template-file, template, templatefile, jsonpath,
	jsonpath-as-json, jsonpath-file).

    --port=-1:
	The port that this container exposes.

    -r, --replicas=1:
	Number of replicas to create. Default is 1.

    --save-config=false:
	If true, the configuration of current object will be saved in its annotation. Otherwise, the annotation will
	be unchanged. This flag is useful when you want to perform kubectl apply on this object in the future.

    --show-managed-fields=false:
	If true, keep the managedFields when printing objects in JSON or YAML format.

    --template='':
	Template string or path to template file to use when -o=go-template, -o=go-template-file. The template format
	is golang templates [http://golang.org/pkg/text/template/#pkg-overview].

    --validate='strict':
	Must be one of: strict (or true), warn, ignore (or false). 		"true" or "strict" will use a schema to validate
	the input and fail the request if invalid. It will perform server side validation if ServerSideFieldValidation
	is enabled on the api-server, but will fall back to less reliable client-side validation if not. 		"warn" will
	warn about unknown or duplicate fields without blocking the request if server-side field validation is enabled
	on the API server, and behave as "ignore" otherwise. 		"false" or "ignore" will not perform any schema
	validation, silently dropping any unknown or duplicate fields.

Usage:
  kubectl create deployment NAME --image=image -- [COMMAND] [args...] [options]

Use "kubectl options" for a list of global command-line options (applies to all commands).
```
- The Validate descriptions is unorganized due to extra spaces.
- image and port description can be improved, we can explain what `-1` signifies.

**Comment by Ritikaa96** [comment]
/sig cli
I would like to work on it if it's triage is accepted.

**Comment by mpuckett159** [comment]
/triage accepted
Can you please update the title to say `kubectl create deployment --help`
I agree that the text for port is confusing and unclear, however I don't think we need to necessarily update text for image. We could add an additional example to detail how the list of images need to be passed in.

**Comment by Ritikaa96** [comment]
> /triage accepted Can you please update the title to say `kubectl create deployment --help` I agree that the text for port is confusing and unclear, however I don't think we need to necessarily update text for image. We could add an additional example to detail how the list of images need to be passed in.

My bad, Thanks for pointing the title @mpuckett159. 
Regarding the image part, I'll add some suggestions in PR. If they don't feel right, feel free to drop a review. It is short and direct , yes. Maybe I'll leave it as is. however something seems missing here.

**Comment by Ritikaa96** [comment]
FYI @mpuckett159 , I have Raised PR: https://github.com/kubernetes/kubernetes/pull/120393 for the issue .

**Comment by Ritikaa96** [comment]
/close
Closed in reference to #120393
